contents/architecture/RESTArchitecturalStyle.md

```diff
@@ -0,0 +1,226 @@
+			<p>For example, the student list of CS3281 can be named as &quot;CS3281StudentList&quot;, and hence can be a resource. The student list might change over time, but the identifier &quot;CS3281StudentList&quot; always refers to the student list regardless of its state/value. </p>
+		</td>
+		<td rowspan="2">
+			Benefits:
+			<ul> 
+				<li> 
+					provides generality by including many sources of information without needing to distinguish their types or implementation 
+				</li> 
+				<li> 
+					free the server from managing different states of resources across requests 
+				</li> 
+				<li> 
+					different <a href="https://en.wikipedia.org/wiki/Media_type#Common_examples">data formats</a> of the representation are available 
+				</li> 
+				<li> 
+					allows <a href="https://en.wikipedia.org/wiki/Late_binding">dynamic binding</a> of the reference to a representation of the resource, hence allowing the server to serve the representation as specified in the request 
+				</li> 
+				<li> 
+					reference the resource instead of its representation saves the trouble of changing the identifier (i.e. the name) when the representation of the resource changes 
+				</li> 
+			</ul>
+		</td>
+	</tr>
+	<tr>
+		<td>
+			Manipulation of resources through representations
+		</td>
+		<td>
+			<p>Representation of resource: &quot;a sequence of bytes, plus the representation metadata to describe those bytes&quot;. The representation of the resource captures the state of the resource, and is transferred between the architectural components. </p>
+			<p>For example, the client requests for a HTML version of &quot;CS3281StudentList&quot; at the start of the semester, the representation of the student list, which includes a string of names and the metadata telling the client that the message is a string of names to be formatted in HTML, will be sent back to the client in the response. </p>
+		</td>
+	</tr>
+	<tr>
+		<td>
+			Self-descriptive messages
+		</td>
+		<td>
+		</td>
+		<td>
+			To support the intermediate processing of interactions. This is related to Stateless and Layered System architectural constraints.
+		</td>
+	</tr>
+	<tr>
+		<td>
+			Hypermedia as the engine of application state (HATEOAS)
+		</td>
+		<td>
+		</td>
+		<td>
+			The client can use the links provided by the server in the response to dynamically navigate to related resources. Benefits:
+			<ul> 
+				<li> 
+					prevent hardcoding of links 
+				</li> 
+				<li> 
+					the server does not drive the change. Hence, the server does not need to store the state of resources across different requests 
+				</li> 
+			</ul>
+		</td>
+	</tr>
+</table>
+
+**Appendix: Summary of Part of Chapter 2**
+
+Architectural properties of key interest in general:
```

Comment:
What is key interest? Instead of having an appendix section, why not have a section for other related resources so that this book chapter is more in line with what the entire learning resources sets out to accomplish?

Reply:
Same as @joanneong , a bit confuse about the term "key interest". In addition, are the following only refer to REST architecture? Or they are traits that all architectures should follow?
Since REST is an alternative to SOAP, can you provide the advantages of REST over SOAP as well?